src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +67,40 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        private async Task CheckHealthEndpointsAsync()
```

Comment:
Since we do not only make requests to the health endpoint, I would suggest renaming this to something like `CheckNetworkHealthAsync`